src/sentry/api/endpoints/monitor_details.py

```diff
@@ -73,14 +92,20 @@ def put(self, request: Request, project, monitor) -> Response:
 
         return self.respond(serialize(monitor, request.user))
 
-    def delete(self, request: Request, project, monitor) -> Response:
+    def delete(
+        self, request: Request, project, monitor, organization_slug: Optional[str] = None
+    ) -> Response:
         """
         Delete a monitor
         ````````````````
 
         :pparam string monitor_id: the id of the monitor.
         :auth: required
         """
+        if organization_slug:
+            if project.organization.slug != organization_slug:
+                return self.respond_invalid()
```

Comment:
I'd like us to think about a way to abstract out these three duplicated lines, especially across other, similar endpoints. A utility function for the slug comparison is a good start. It would be even better if we can get rid of the "if...return" by raising an exception to trigger the 400 response. To generalize across other endpoints, a mixin superclass might be the way to go. We can work on this later -- just something to think about for now.